src/main_app/components/SignUp/index.js

```diff
@@ -0,0 +1,100 @@
+import { useState } from 'react'
+import { loader } from 'graphql.macro'
+import { Formik, Field, Form, ErrorMessage } from 'formik'
+import * as Yup from 'yup'
+import { useMutation, Mutation } from '@apollo/client'
+
+import client from 'services/graphql/client'
+
+const sleep = (ms) => new Promise((r) => setTimeout(r, ms))
+
+const exampleMutation = loader('../../graphql/createUser.graphql')
+
+const SignupSchema = Yup.object().shape({
+  firstName: Yup.string()
+    .min(4, 'Too Short!')
+    .max(50, 'Too Long!')
+    .required('Required'),
+  lastName: Yup.string()
+    .min(2, 'Too Short!')
+    .max(50, 'Too Long!')
+    .required('Required'),
+  email: Yup.string().email('Invalid email').required('Required'),
+  password: Yup.string()
+    .min(8, 'Too Short!')
+    .max(12, 'Too Long!')
+    .required('Required')
+})
+
+const SignUp = () => {
+  const [state, setState] = useState ({
+    firstName: "",
+    lastName: "",
+    email: "",
+    password: ""
+  })
+  const { loading, error, data } = useMutation(exampleMutation, { client })
```

Comment:
`useMutation` returns an array with the function to execute the mutation in the first position. You need to extract that function to run the mutation:

```suggestion
  const [runExampleMutation] = useMutation(exampleMutation, { client })
```

Then, in the `onSubmit` event from Formik, call this function to execute the mutation:

```javascript
onSubmit={async (values, { setSubmitting }) => {
  try {
    const { data } =await runExampleMutation({ variables: values })
    alert(JSON.stringify(data))
  } catch (err) {
    alert(err.mesage)
  } finally {
    setSubmitting(false)
  }
}}
```